Xabe.AutoUpdater/FtpProvider.cs

```diff
@@ -0,0 +1,470 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.IO.Compression;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Xml.Serialization;
+using Octokit;
+using FileMode = System.IO.FileMode;
+
+namespace Xabe.AutoUpdater
+{
+    /// <summary>
+    ///     Provider for FtpReleases
+    /// </summary>
+    public class FtpProvider : IReleaseProvider
+    {
+        private readonly string _appName;
+        private readonly string _ftpHosts;
+        private readonly string _ftpLogin;
+        private readonly string _ftpPassword;
+        private readonly string _ftpFolder;
+        private readonly bool _ftpPassiveMode;
+
+        /// <summary>
+        ///     Create instance of FtpProvider. 
+        /// </summary>
+        /// <param name="appName">Name of current application.</param>
+        /// <param name="pHosts">FTP HOST.</param>
+        /// <param name="pLogin">FTP Account Login</param>
+        /// <param name="pPassword">FTP Account Password</param>
+        /// <param name="pPassiveMode">FTP Passive Mode</param>
+        /// <param name="pFolder">FTP Release Folder (optional)</param>
+        public FtpProvider(string appName, string pHosts, string pLogin, string pPassword, bool pPassiveMode, string pFolder = null)
+        {
+            _appName = appName;
+            _ftpHosts = pHosts;
+            _ftpLogin = pLogin;
+            _ftpPassiveMode = pPassiveMode;
+            _ftpPassword = pPassword;
+            _ftpFolder = string.IsNullOrEmpty(pFolder) ? "/" : pFolder;
+        }
+
+        /// <inheritdoc />
+        public Task<List<string>> DownloadCurrentVersion()
+        {
+            var t = Task.Run(async () =>
+            {
+                var client = BuildClient();
+                var latestVersion = await GetLatestVersionNumber();
+                if (!string.IsNullOrEmpty(latestVersion))
+                {
+                    var remoteFile = string.Concat(_ftpFolder, $"{_appName}.{latestVersion}.zip");
```

Comment:
You should find latest version by comparing versions from all files in directory.